influxdb_retention_policy: don't fail with empty error message

agx opened the pull request:
##### SUMMARY
On connection errors the influxdb python module raises an exception with
an empty message leading to a fail_json with an empty error message:

    https://github.com/influxdata/influxdb-python/blob/265d14736bb739aed0f357d8da248d51f549244b/influxdb/influxdb08/client.py#L252

Fail with a more descriptive error in this case.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
influxdb_retention_policy

##### ANSIBLE VERSION
```
ansible 2.4.0 (influxdb-connection-error daad4b3d86) last updated 2017/08/29 20:12:04 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/agx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /var/scratch/src/ansible/ansible/lib/ansible
  executable location = /var/scratch/src/ansible/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```

resmo commented:
shipit

agx commented:
I think this needs a second shipit to move further?

Akasurde commented on lib/ansible/modules/database/influxdb/influxdb_retention_policy.py:
```diff
@@ -156,7 +157,7 @@ def find_retention_policy(module, client):
                 break
     except requests.exceptions.ConnectionError as e:
-        module.fail_json(msg=str(e))
+        module.fail_json(msg=(str(e) or "cannot connect to database %s on %s" % (database_name, hostname)))
```
Can you replace str with to_native ?

```
msg="Cannot connect to database %s on %s : %s" % (database_name, hostname, to_native(e))
```

agx committed:
influxdb_retention_policy: don't fail with empty error message

On connection errors the influxdb python module raises an exception with
an empty message leading to a fail_json with an empty error message:

    https://github.com/influxdata/influxdb-python/blob/265d14736bb739aed0f357d8da248d51f549244b/influxdb/influxdb08/client.py#L252

Fail with a more descriptive error in this case.

agx commented:
@Akasurde updated! Thanks for having a look

Akasurde commented on lib/ansible/modules/database/influxdb/influxdb_retention_policy.py:
```diff
@@ -115,7 +116,7 @@ def find_retention_policy(module, client):
                 break
     except requests.exceptions.ConnectionError as e:
-        module.fail_json(msg=str(e))
+        module.fail_json(msg=("Cannot connect to database %s on %s : %s" % (database_name, hostname, to_native(e))))
```
Remove redundant round brackets after = sign

```
msg="Cannot connect to database %s on %s : %s" % (database_name, hostname, to_native(e))
```

agx commented:
@Akasurde done.

Akasurde commented:
LGTM, @resmo could you please take a look again ?

resmo commented:
shipit

mkrizek commented:
rebuild_merge
